zebra-chain/src/block.rs

```diff
@@ -86,6 +87,34 @@ impl Block {
             Some(height) => Commitment::from_bytes(self.header.commitment_bytes, network, height),
         }
     }
+
+    /// Check if the `branch_id` field from the transactions match block height.
+    ///
+    /// # Consensus rule:
+    ///
+    ///  The nConsensusBranchId field MUST match the consensus branch ID used for
+    ///  SIGHASH transaction hashes, as specifed in [ZIP-244] ([7.1]).
+    ///
+    /// [ZIP-244]: https://zips.z.cash/zip-0244
+    /// [7.1]: https://zips.z.cash/protocol/nu5.pdf#txnencodingandconsensus
+    pub fn check_consensus_branch_id_consistency(
```

Comment:
This method name will be confusing, because the field is called `network_upgrade`
```suggestion
    pub fn check_transaction_network_upgrades(
```

Reply:
https://github.com/ZcashFoundation/zebra/pull/2354/commits/1c0adc621b67e53d07ea6b5420cd637ed34dab89